cmd/kube-controller-manager/app/cloudproviders.go

```diff
@@ -32,6 +34,13 @@ func createCloudProvider(cloudProvider string, externalCloudVolumePlugin string,
 	var cloud cloudprovider.Interface
 	var loopMode ControllerLoopMode
 	var err error
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) && cloudprovider.IsDeprecatedInternal(cloudProvider) {
```

Comment:
Why doesn't the cloud-controller-manager also have a block like this?

Reply:
That's a good question. @cheftako @andrewsykim, should this FeatureGate also disable cloud providers in depricated in-tree CCM? Enhancement only covered KAPI, KCM and Kubelet - https://github.com/kubernetes/enhancements/pull/2443/files#diff-9df64bb76071963721a9e1df8a7c7a4e838d96772d8f1c0951d2d8bb9246608cR117

Reply:
cloud-controller-manager in the k/k repo is just a reference example used by external cloud providers, so it shouldn't have to check this feature gate

Reply:
The CCM's job is to host cloud providers. The flag is a way to disable the cloud provider from the places we intend to remove it. So I believe the flag should be checked in the KAS, the KCM and the Kubelet. It should not be checked in the CCM and in part that is ok, because as Andrew points out this is a reference implementation and not shipped with K8s. The actual implementations should not use it either and I do not want to accidentally have a Cloud Provider pick up that flag.